akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -2212,14 +2212,20 @@ private[akka] final class StatefulMapConcat[In, Out](val f: () => In => Iterable
     lazy val decider = inheritedAttributes.mandatoryAttribute[SupervisionStrategy].decider
     var currentIterator: Iterator[Out] = _
     var plainFun = f()
+    val contextPropagation = ContextPropagation()
 
     def hasNext = if (currentIterator != null) currentIterator.hasNext else false
 
     setHandlers(in, out, this)
 
-    def pushPull(): Unit =
+    def pushPull(pullFirstSubElement: Boolean): Unit =
```

Comment:
Maybe just rename the flag to `shouldResumeContext`.

Reply:
Or to keep the original "polarity" of the flag: `usingOriginalContext`.

Then the condition would read `if(!usingOriginalContext) contextPropagation.resumeContext()`

Reply:
I think `shouldResumeContext` better stands the intention